api/messages.proto

```diff
@@ -34,10 +35,28 @@ message Ticket {
 
   // Properties contains custom info about the ticket.  Top level values can be
   // used in indexing and filtering to find tickets.
+  // TODO: Deprecate and remove this field.
   google.protobuf.Struct properties = 2;
 
   // Assignment associated with the Ticket.
   Assignment assignment = 3;
+
+  SearchFields search_fields = 4;
+
+  google.protobuf.Any criteria = 5;
```

Comment:
Naming is hard :) - 'criteria' is confusing. I think 'properties' seems to be the closest - we should be ok making braking changes at this stage right?

Reply:
How about "mmf_input"?  I want to keep it clear that this is that any for data for helping run the mmf, and not as input for other systems (which would need their own fields to preserve system composability.)

Changed some other fields to match, not really happy with "handling_output".

Reply:
Changed "handling_output" fields to "fe_outout" and "be_output", better at least.